src/libraries/Microsoft.PowerFx.Connectors/Environment/PowerFxConfigExtensions.cs

```diff
@@ -22,24 +24,36 @@ public static class ConfigExtensions
         /// <param name="config">Config to add the functions to.</param>
         /// <param name="connectorSettings">Connector settings containing Namespace and MaxRows to be returned.</param>        
         /// <param name="openApiDocument">An API document. This can represent multiple formats, including Swagger 2.0 and OpenAPI 3.0.</param>
-        public static IReadOnlyList<ConnectorFunction> AddActionConnector(this PowerFxConfig config, ConnectorSettings connectorSettings, OpenApiDocument openApiDocument)
+        /// <param name="configurationLogger">Logger.</param>
+        /// <returns>List of connector functions.</returns>
+        public static IReadOnlyList<ConnectorFunction> AddActionConnector(this PowerFxConfig config, ConnectorSettings connectorSettings, OpenApiDocument openApiDocument, ConnectorLogger configurationLogger = null)
         {
-            if (connectorSettings.Namespace == null)
-            {
-                throw new ArgumentNullException(nameof(connectorSettings.Namespace));
-            }
+            Guid id = ConnectorLogger.NewId();            
```

Comment:
>  Guid id = ConnectorLogger.NewId();             [](http://example.com/codeflow?start=10&length=48)

from an encapsulation perspective, consider encapsulating the id. 

var logScope = configurationLogger?.NewScope(); // this gets the id

logScope.LogInformation("message",..); // no id needed here.